### PR TITLE
(LTH-135) Populate locale file location based on install location

### PR DESCRIPTION
**Pull request by Magisus**
This commit updates the `gettext_compile` helper to use Leatherman's
install location as a reference point for translation file location.

### CODE REVIEW

**Comment by Magisus** [comment]
When we are ready for 1.0, we can remove the `inst` argument to `gettext_compile`.

File: CMakeLists.txt [hunk 1]
```diff
@@ -2,6 +2,8 @@ cmake_minimum_required(VERSION 3.2.2)
 project(leatherman VERSION 0.12.0)
 
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
+# Populate locale install location
+configure_file(cmake/leatherman.cmake.in "${CMAKE_CURRENT_SOURCE_DIR}/cmake/leatherman.cmake" @ONLY)
```
Comment (MikaelSmith) [inline CMakeLists.txt]: I wish there were a better way. However this all seems like it should work.

**Comment by MikaelSmith** [comment]
@branan do you have any better ideas on how to do this? You had more of a hand in the Leatherman config installed files.

**Comment by branan** [comment]
This seems like the best option to me, off the top of my head

File: CMakeLists.txt (hunk 1 above)
Comment (MikaelSmith) [inline CMakeLists.txt]: I'd rather this were written into CMAKE_BINARY_DIR (and we added that to the CMAKE_MODULE_PATH and installed from there as well).

We haven't been good about that for generated files, but it makes it difficult to have more than one build of the same project. I'd like to move towards cleaner isolation of generated files.

Comment (Magisus) [inline CMakeLists.txt]: So I see us using PROJECT_BINARY_DIR here: https://github.com/puppetlabs/leatherman/blob/master/CMakeLists.txt#L125-L126. According to the docs, CMAKE_BINARY_DIR is the top level of build tree (i.e. == CMAKE_SOURCE_DIR in our use case), while PROJECT_BINARY_DIR is the build directory. Are the two file generation use cases here different and therefore should use different targets, or did you mean one or the other?

Comment (MikaelSmith) [inline CMakeLists.txt]: I mean PROJECT_BINARY_DIR.

Comment (MikaelSmith) [inline CMakeLists.txt]: I don't see the others being used during the Leatherman build process, so leatherman.cmake might need to go in a subdirectory of PROJECT_BINARY_DIR so you can safely add it to CMAKE_MODULE_PATH.

Comment (Magisus) [inline CMakeLists.txt]: Cool, that makes sense to me.

**Comment by Magisus** [comment]
Updated, realized the special casing was actually broken, since LEATHERMAN_LOCALE_INSTALL is always set here https://github.com/puppetlabs/leatherman/blob/master/CMakeLists.txt#L44-L51. If LEATHERMAN_LOCALE_INSTALL is a relative path, we want to prepend Leatherman's CMAKE_INSTALL_PREFIX. If it's absolute, that means the install prefix has already been added, and we don't want to add it again.

**Comment by Magisus** [comment]
Note this assumes that LEATHERMAN_LOCALE_VAR points to the Leatherman install root. If it does not, this is still broken. Do we want to ignore that case, or does someone have an idea for how to account for it? I've been having trouble getting the environment variables to be interpolated right...

**Comment by MikaelSmith** [comment]
@Magisus the thing you pointed to was adding a C++ define, it wouldn't have caused the variable to be set in CMAKE. So I think the original worked.

**Comment by Magisus** [comment]
Annnd yeah. You're totally right. Too much wading through the weeds... >.<

**Comment by Magisus** [comment]
So, I verified that as-is, this works in puppet-agent on Windows with no configuration needed for any component besides Leatherman, including when the user changes the default installation directory using the Install Wizard.

However, this will NOT work for the general case of Leatherman + consuming project, if `LEATHERMAN_LOCALE_VAR` is an arbitrary location, and not the Leatherman install root. Neither Leatherman nor its consuming projects will be able to find their translation files in this case. If we're okay with this, I'm happy to have this merged. If not, we will have to consider either other approaches, or change the use case, e.g. by rewording the readme to indicate that the `LEATHERMAN_LOCALE_VAR` should only ever be used to point to the Leatherman install root, instead of the arbitrary example currently there.

**Comment by Magisus** [comment]
Also note that in some cases, it might necessary for consuming projects to set `LEATHERMAN_LOCALE_INSTALL`, it just doesn't happen to be needed (by sheer luck) in puppet-agent.

**Comment by MikaelSmith** [comment]
I think for now let's go with changing the README to be more restrictive about the use case. In the future it may make sense to switch to using the registry, as using an environment variable is pretty restrictive; as such I don't think it's worth re-architecting to make the environment variable option work more broadly.

**Comment by Magisus** [comment]
Works for me, Readme is updated.

File: cmake/leatherman.cmake.in [hunk 2]
```diff
@@ -111,7 +116,7 @@ macro(enable_cppcheck)
 endmacro()
 
 # We set this here so that enable_cpplint() can find it
-set(LEATHERMAN_CPPLINT_PATH "${CMAKE_CURRENT_LIST_DIR}/../scripts/cpplint.py")
+set(LEATHERMAN_CPPLINT_PATH "${CMAKE_CURRENT_LIST_DIR}/../../scripts/cpplint.py")
```
Comment (MikaelSmith) [inline cmake/leatherman.cmake.in]: This seems like it'd be similar to treatment of `LEATHERMAN_CMAKE_DIR`. Why isn't it the same problem?

Comment (Magisus) [inline cmake/leatherman.cmake.in]: I think it is the same problem, but I found this one first. I'll refactor it to use `LEATHERMAN_CMAKE_DIR`.

**Pull request merged**